### PR TITLE
fix: pptx uploaded on google docs 

### PR DESCRIPTION
**Pull request by milovate**
### Q/A checklist

- [ ] If you add new dependencies, did you update the lock file?
```bash
poetry lock --no-update
```
- [ ] Run tests 
```bash
ulimit -n unlimited && ./scripts/run-tests.sh
```
- [ ] Do a self code review of the changes - Read the diff at least twice.  
- [ ] Carefully think about the stuff that might break because of this change - this sounds obvious but it's easy to forget to do "Go to references" on each function you're changing and see if it's used in a way you didn't expect. 
- [ ] The relevant pages still run when you press submit
- [ ] The API for those pages still work (API tab)
- [ ] The public API interface doesn't change if you didn't want it to (check API tab > docs page)
- [ ] Do your UI changes (if applicable) look acceptable on mobile?
- [ ] Ensure you have not regressed the import time unless you have a good reason to do so. 
You can visualize this using tuna:
```bash
python3 -X importtime -c 'import server' 2> out.log && tuna out.log
```
To measure import time for a specific library:
```bash
$ time python -c 'import pandas'

________________________________________________________
Executed in    1.15 secs    fish           external
   usr time    2.22 secs   86.00 micros    2.22 secs
   sys time    0.72 secs  613.00 micros    0.72 secs
```
To reduce import times, import libraries that take a long time inside the functions that use them instead of at the top of the file:
```python
def my_function():
    import pandas as pd
    ...
```

### Legal Boilerplate

Look, I get it. The entity doing business as “Gooey.AI” and/or “Dara.network” was incorporated in the State of Delaware in 2020 as Dara Network Inc. and is gonna need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Dara Network Inc can use, modify, copy, and redistribute my contributions, under its choice of terms.

### CODE REVIEW

**Comment by devxpy** [comment]
I think we talked about exporting the file as presentation in the first place here
https://github.com/GooeyAI/gooey-server/blob/66a05d78bd85312747343e6b976b1c75f386af92/daras_ai_v2/gdrive_downloader.py#L159-L161

I would even go further to suggest that you likely don't need export_media() at all for presentation and just use get_media() always

**Comment by devxpy** [comment]
> I think we talked about exporting the file as presentation in the first place here
> 
> https://github.com/GooeyAI/gooey-server/blob/66a05d78bd85312747343e6b976b1c75f386af92/daras_ai_v2/gdrive_downloader.py#L159-L161

Ok nevermind, I don't think that works. But we can be smart about it by looking at the mime type returned by `gdrive_metadata` (which we already have here)

**Comment by devxpy** [comment]
In addition, we can also add `exportLinks` to the `gdrive_metadata` to get export slides larger than 10mb https://stackoverflow.com/a/59168288/7061265

File: daras_ai_v2/gdrive_downloader.py [hunk 1]
```diff
@@ -60,16 +62,27 @@ def gdrive_list_urls_of_files_in_folder(f: furl, max_depth: int = 4) -> list[str
...
+    request, mime_type = service_request(service, file_id, f, mime_type)
+    file_bytes, mime_type = download_blob_file_content(
+        service, request, file_id, f, mime_type, export_links
+    )
+
+    return file_bytes, mime_type
+
+
+def service_request(
+    service, file_id: str, f: furl, mime_type: str, retried_request=False
```
Comment (devxpy) [inline daras_ai_v2/gdrive_downloader.py:82]: remove unused param

File: daras_ai_v2/gdrive_downloader.py [hunk 2]
```diff
@@ -81,17 +94,60 @@ def gdrive_download(f: furl, mime_type: str) -> tuple[bytes, str]:
...
+            # print(f"Download {int(status.progress() * 100)}%")
+        f_bytes = file.getvalue()
+
     return f_bytes, mime_type
 
 
+def download_from_exportlinks(f: furl) -> bytes:
+    try:
+        r = requests.get(f)
+        f_bytes = r.content
```
Comment (devxpy) [inline daras_ai_v2/gdrive_downloader.py:145]: use raise_for_status() like we do everywhere else in code

Comment (devxpy) [reply daras_ai_v2/gdrive_downloader.py:145]: this likely doesnt support downloading private docs without auth, I may be wrong

File: daras_ai_v2/gdrive_downloader.py [hunk 3]
```diff
@@ -81,17 +94,60 @@ def gdrive_download(f: furl, mime_type: str) -> tuple[bytes, str]:
...
+                # print(f"Download {int(status.progress() * 100)}%")
+            f_bytes = file.getvalue()
+
+    else:
+        done = False
+        while done is False:
+            _, done = downloader.next_chunk()
+            # print(f"Download {int(status.progress() * 100)}%")
+        f_bytes = file.getvalue()
+
```
[the remark this answers is not in the document]
Comment (milovate) [reply daras_ai_v2/gdrive_downloader.py:138]: ah! , yes this make way more sense, i wanted to only use export_links for pptx without breaking the existing logic hence the mess.

**Pull request merged**